bin/critterget.py

```diff
@@ -563,19 +563,23 @@ def getCredentials(sessionKey):
 
     if (debug) : print u'{} MessageType="ApteligentDebug"  Into getCredentials'.format(DATETIME_OF_RUN)
 
-    try:
-        # list all credentials
-        entities = entity.getEntities(['admin', 'passwords'], namespace=myapp,
-                                    owner='nobody', sessionKey=sessionKey)
-    except Exception, e:
-        print u'{} MessageType="ApteligentDebug" Could not get {} credentials from splunk. Error: {}'.format(DATETIME_OF_RUN, myapp, str(e))
-
-    # return first set of credentials
-    if (debug) : print "Entities is ", entities
-    for i, c in entities.items():
-        return c['clear_password']
-
-    print u'{} MessageType="ApteligentDebug" No credentials have been found for app {} . Maybe a setup issue?'.format(DATETIME_OF_RUN, myapp)
+    auth = None
+
+    while auth is None:
```

Comment:
This is potentially an infinite loop / infinite retry.  Is that really what you want?  There should be some limit, after which you give up.  No more than 10 IMO.  Also, when you print the error message, print which try number you're on.

If you want to get fancier, sleep a bit between retries.  If you want to get *really* fancy, do an exponential backoff between retries.